src/Illuminate/View/Compilers/Concerns/CompilesComponents.php

```diff
@@ -150,7 +150,10 @@ protected function compileEndComponentFirst()
      */
     protected function compileProps($expression)
     {
-        return "<?php \$attributes = \$attributes->exceptProps{$expression}; ?>
+        return "<?php foreach(\$attributes->onlyProps{$expression} as \$__key => \$__value) {
```

Comment:
For classless components using the directive `@prop`, this piece of code extracts from `$attributes` the props and declares them for the view.